python/GafferSceneUI/SetEditor.py

```diff
@@ -0,0 +1,379 @@
+		result = []
+		for p in selection.paths() :
+			path.setFromString( p )
+			setName = path.property( "setPath:setName" )
+			if setName is not None :
+				result.append( setName )
+
+		return result
+
+	def __dragBegin( self, widget, event ) :
+
+		path = self.__pathListing.pathAt( imath.V2f( event.line.p0.x, event.line.p0.y ) )
+		column = self.__pathListing.columnAt( imath.V2f( event.line.p0.x, event.line.p0.y ) )
+		selection = self.__pathListing.getSelection()
+
+		if selection.match( str( path ) ) & IECore.PathMatcher.Result.ExactMatch :
+			if column == self.__setMembersColumn :
+				GafferUI.Pointer.setCurrent( "paths" )
+				return IECore.StringVectorData( self.__getSetMembers().paths() )
+			else :
+				selectedSetNames = self.__selectedSetNames()
+				if len( selectedSetNames ) > 0 :
+					GafferUI.Pointer.setCurrent( "paths" )
+					return IECore.StringVectorData( selectedSetNames )
+				else :
+					# prevent the path itself from being dragged
+					return IECore.StringVectorData()
+
+		return None
+
+	def __keyPressSignal( self, widget, event ) :
+
+		if event.key == "C" and event.modifiers == event.Modifiers.Control :
+			self.__copySelectedSetNames()
+			return True
+		elif event.key == "C" and event.modifiers == event.Modifiers.ShiftControl :
+			self.__copySetMembers()
+			return True
+
+		return False
+
+	def __contextMenuSignal( self, widget ) :
+
+		menuDefinition = IECore.MenuDefinition()
+
+		selection = self.__pathListing.getSelection()
+		selectedSetNames = self.__selectedSetNames()
+
+		menuDefinition.append(
+			"/Copy Set Name%s" % ( "" if selection.size() == 1 else "s" ),
+			{
+				"command" : Gaffer.WeakMethod( self.__copySelectedSetNames ),
+				"active" : len( selectedSetNames ) > 0,
+				"shortCut" : "Ctrl+C"
+			}
+		)
+
+		menuDefinition.append(
+			"/Copy Set Members",
+			{
+				"command" : Gaffer.WeakMethod( self.__copySetMembers ),
+				"active" : len( selectedSetNames ) > 0,
+				"shortCut" : "Ctrl+Shift+C"
+			}
+		)
+
+		menuDefinition.append(
+			"/Select Set Members",
+			{
+				"command" : Gaffer.WeakMethod( self.__selectSetMembers ),
+				"active" : len( selectedSetNames ) > 0,
+			}
+		)
+
+		self.__contextMenu = GafferUI.Menu( menuDefinition )
+		self.__contextMenu.popup( widget )
+
+		return True
+
+	def __copySelectedSetNames( self, *unused ) :
+
+		if self.__plug is None :
+			return
+
+		selection = self.__selectedSetNames()
+		if len( selection ) > 0 :
+			data = IECore.StringVectorData( selection )
+			self.__plug.ancestor( Gaffer.ApplicationRoot ).setClipboardContents( data )
+
+	def __getSetMembers( self, *unused ) :
+
+		if self.__plug is None :
+			return
+
+		result = IECore.PathMatcher()
+		with Gaffer.Context( self.getContext() ) :
+			for setName in self.__selectedSetNames() :
+				result.addPaths( self.__plug.set( setName ).value )
+
+		return result
+
+	def __selectSetMembers( self, *unused ) :
+
+		locations = self.__getSetMembers()
+		if not locations.isEmpty() :
```

Comment:
I feel like it might be worth dropping this check, so that no matter what, after the action is complete, the current selection is exactly the same as the content of the set. I think confusion could ensue if you quickly use the shortcut without looking at the members column, and then assume that the current selection (the old one that was left unchanged) is in the set.

Reply:
Sounds good to me. Changed selection behaviour in c133edd. Also gave the copy actions the same behaviour in b98dd93.